### ISSUE TITLE
rs

### ISSUE THREAD

**Issue by canstudios-louisem** [adaptlearning/adapt_authoring#1705]
Currently, our test coverage is quite poor, here is the coverage calculation by [NYC](https://www.npmjs.com/package/nyc):

```
Statements   : 45.15% ( 2737/6062 )
Branches: 30.61% ( 888/2901 )
Functions: 45.55% ( 584/1282 )
Lines: 45.27% ( 2736/6044 )
```

I think it is actually lower than this as there are files tests simply don't touch.

In my opinion, we should be a lot stricter about tests being included in pr's.

A tool I have used in the past is https://coveralls.io/ which I found quite helpful to track coverage and make sure any pr's or features didn't make things worse.

I don't think we need to be chasing 100% you just end up writing tests for the sake of coverage but 80-90% would give us a lot more confidence and would allow us to leverage other tools that run off the ci more effectivly.

**Comment by taylortom** [comment]
Yeah, as mentioned on the call these were added initially and not kept up-to-date.

I suggest we plan a working group to look into all unit tests after the 0.4.1 patch release, and alongside tenant management and import/export; I'll have time to donate to this as I'm not involved in those.

**Comment by canstudios-louisem** [comment]
Works for me I just wanted to get something down while it was in my head.
